test/smartwallet/nativeHolderSmartWallet.test.ts

```diff
@@ -0,0 +1,262 @@
+import { use, expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import { BN, bufferToHex, privateToAddress, toBuffer } from 'ethereumjs-util';
+import { toWei } from 'web3-utils';
+import {
+    NativeHolderSmartWalletInstance,
+    SmartWalletFactoryInstance,
+    TestRecipientInstance,
+    TestTokenInstance
+} from '../../types/truffle-contracts';
+import { constants } from '../constants';
+import {
+    createRequest,
+    createSmartWallet,
+    createSmartWalletFactory,
+    generateBytes32,
+    getGaslessAccount,
+    getTestingEnvironment,
+    signRequest
+} from '../utils';
+
+use(chaiAsPromised);
+
+const NativeSmartWallet = artifacts.require('NativeHolderSmartWallet');
+const TestToken = artifacts.require('TestToken');
+const TestRecipient = artifacts.require('TestRecipient');
+
+contract.only('NativeHolderSmartWallet', ([worker, fundedAccount]) => {
```

Comment:
we should remove `only` from here

Reply:
sorry, my bad I still need to add the missing tests but I wanted to check functionality